src/main/java/org/apache/flink/benchmark/CheckpointingTimeBenchmark.java

```diff
@@ -97,43 +100,119 @@
                         .build();
 
         new Runner(options).run();
     }
 
     @Benchmark
-    public void debloatedCheckpointSingleInput(DebloatedCheckpointEnvironmentContext context)
+    public void checkpointSingleInput(DebloatedCheckpointEnvironmentContext context)
             throws Exception {
         final CompletableFuture<String> checkpoint =
                 context.miniCluster.triggerCheckpoint(context.jobID);
         checkpoint.get();
     }
 
+    public enum CheckpointingMode {
+        UNALIGNED(
+                config -> {
+                    config.set(ExecutionCheckpointingOptions.ENABLE_UNALIGNED, true);
+                    config.set(TaskManagerOptions.MEMORY_SEGMENT_SIZE, START_MEMORY_SEGMENT_SIZE);
+                    config.set(
+                            ExecutionCheckpointingOptions.ALIGNED_CHECKPOINT_TIMEOUT,
+                            Duration.ofMillis(0));
+                    config.set(TaskManagerOptions.BUFFER_DEBLOAT_ENABLED, false);
+                    return config;
+                },
+                () -> null,
+                UNALIGNED_RECORD_SIZE),
+        ALTERNATING(
+                config -> {
+                    config.set(ExecutionCheckpointingOptions.ENABLE_UNALIGNED, true);
+                    config.set(TaskManagerOptions.MEMORY_SEGMENT_SIZE, START_MEMORY_SEGMENT_SIZE);
+                    config.set(
+                            ExecutionCheckpointingOptions.ALIGNED_CHECKPOINT_TIMEOUT,
+                            Duration.ofMillis(1));
+                    config.set(TaskManagerOptions.BUFFER_DEBLOAT_ENABLED, false);
+                    return config;
+                },
+                () -> null,
+                UNALIGNED_RECORD_SIZE),
+        ALIGNED(
+                config -> {
+                    config.set(ExecutionCheckpointingOptions.ENABLE_UNALIGNED, false);
+                    config.set(TaskManagerOptions.MEMORY_SEGMENT_SIZE, START_MEMORY_SEGMENT_SIZE);
+                    config.set(TaskManagerOptions.MIN_MEMORY_SEGMENT_SIZE, MIN_MEMORY_SEGMENT_SIZE);
+                    config.set(TaskManagerOptions.BUFFER_DEBLOAT_ENABLED, true);
+                    config.set(TaskManagerOptions.BUFFER_DEBLOAT_TARGET, DEBLOATING_TARGET);
+                    config.set(
+                            TaskManagerOptions.BUFFER_DEBLOAT_PERIOD,
+                            Duration.of(10, ChronoUnit.MILLIS));
+                    config.set(TaskManagerOptions.BUFFER_DEBLOAT_SAMPLES, 5);
+                    return config;
+                },
+                () -> {
+                    // wait a bit for debloating to stabilize
+                    Thread.sleep(2_000);
+                    return null;
+                },
+                DEBLOATING_RECORD_SIZE);
+
+        private final Function<Configuration, Configuration> configFunc;
+        private final SupplierWithException<Void, Exception> postSetUpFunc;
```

Comment:
`FunctionWithException`? Or even better, why don't you change it to a simple `int setupSleepDelay`?

Reply:
I don't like the `int setupSleepDelay` because it is less flexible, plus I'd need to add an `if (setupSleepDelay > 0)`.

What's wrong with `Supplier`? It's just a function that does not take arguments.

Reply:
> I don't like the int setupSleepDelay because it is less flexible, plus I'd need to add an if (setupSleepDelay > 0).

You could have the same argument in favour of replacing all of the regular function arguments with some lambda functions. Yes, it's less flexibly, but why use more complicated construct for something that could be expressed easily as a single parameter?

> What's wrong with Supplier? It's just a function that does not take arguments.

Readability. Putting aside strange `return null;` that does nothing, in order to understand what is the following line doing:
```
mode.postSetUp();
```
With supplier/function you need to jump through the code much more and this flexibility is actually hurting your understanding, as you need to see a definition of every possible `SupplierWithException<Void, Exception> postSetUpFunc` parameter to understand what is this single line doing in all of the cases. Currently, in this case, it's not that big of a deal, as everything is one file, relatively close to one another, but the point still stands. If you see
```
mode.postSetUp() {
  postSetUpFunc.get();
}
```
it's very non obvious what is this doing. With
```
Thread.sleep(mode.postSetUpSleep);
```
it's much easier to understand.

Reply:
I will not argue about such a minimal change... Will update it, even if I see it differently.